src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -425,11 +425,14 @@ public UnfilteredPartitionIterator executeLocally(ReadExecutionController execut
                                             .collect(Collectors.joining(",")));
             }
 
-            UnfilteredPartitionIterator iterator = (null == searcher) ? queryStorage(cfs, executionController) : searcher.search(executionController);
```

Comment:
I think that this should be moved inside `try` because if we get iterator and then it will throw in `RTBoundValidator.validate` then such iterator is never closed. Might happen ... right?

Reply:
I don't think there's actually anything in `RTBoundValidator.validate()` that can throw, and in the previous line, we'll fail before we even get an iterator. I'd revert this, at least as part of this patch. Even if we really wanted to change it, I think we'd just want to a.) move only `validate()` into the `try` and b.) remove the unnecessary `iterator == null` check.